Jobs/Notifications/IssuePublishedNotifyUsers.php

```diff
@@ -17,44 +17,82 @@
 
 use APP\core\Application;
 use APP\facades\Repo;
+use APP\issue\Issue;
 use APP\mail\mailables\IssuePublishedNotify;
+use APP\notification\Notification;
+use APP\notification\NotificationManager;
 use Illuminate\Bus\Batchable;
 use Illuminate\Support\Collection;
 use Illuminate\Support\Facades\Mail;
+use InvalidArgumentException;
 use PKP\context\Context;
 use PKP\emailTemplate\EmailTemplate;
 use PKP\Support\Jobs\BaseJob;
 use PKP\user\User;
 
-class IssuePublishedMailUsers extends BaseJob
+class IssuePublishedNotifyUsers extends BaseJob
 {
     use Batchable;
 
     protected Collection $recipientIds;
     protected int $contextId;
-    protected User $sender;
+    protected Issue $issue;
     protected string $locale;
 
-    public function __construct(Collection $recipientIds, int $contextId, User $sender, string $locale)
-    {
+    // Sender of the email; should be set if sendEmail is true
+    protected ?User $sender;
+
+    // Whether to send notification email
+    protected bool $sendEmail = false;
+
+    public function __construct(
+        Collection $recipientIds,
+        int $contextId,
+        Issue $issue,
+        string $locale,
+        ?User $sender = null,
+        bool $sendEmail = false
```

Comment:
Maybe too late to bother with, but could these last two arguments be reduced to one? 

```
@param ?User $sender The sender of the email. Leave null to not send an email.
```